### ISSUE TITLE
Warning in browser console: `Canvas2D: Multiple readback operations using getImageData are faster with the willReadFrequently attribute set to true` when calling `toDataURL`

### ISSUE THREAD

**Issue by nmalancea** [agilgur5/react-signature-canvas#105]
![Screenshot 2024-04-29 at 1 01 41 PM](https://github.com/agilgur5/react-signature-canvas/assets/3107683/10aad32f-43bb-4425-819a-e56b4d11bb7a)

I get this warning in the browser console when calling `toDataURL`:

```
Canvas2D: Multiple readback operations using getImageData are faster with the willReadFrequently attribute set to true
```

Any ideas how to correctly configure the canvas to set `{ willReadFrequently: true }`?

**Comment by agilgur5** [comment]
We might need to add a prop for the canvas's [`contextAttributes`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext#contextattributes)

**Comment by Beatriz2529** [comment]
Hello, I have the same problem.
Could you give me an exemple please ?

**Comment by agilgur5** [comment]
I don't believe there's a workaround for this as the `CanvasContext` manipulation is not user-facing.

In any case, this is a warning from a more recent browser feature and can safely be ignored.

**Comment by hossain45** [comment]
Does this error cause any performance issue? since you haven't added the feature yet shall I ignore this error at the moment? @agilgur5

**Comment by agilgur5** [comment]
It's not an error, it's a warning, so by definition you can ignore it.

And the warning is about a newer browser feature that did not exist until years after this library was created. It worked fine without the feature and will continue to work fine without it.

Enabling it for some use-cases will allow for a performance optimization (which is what this issue is labeled as). But again, it works fine without this.

**Comment by agilgur5** [comment]
> We might need to add a prop for the canvas's [`contextAttributes`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext#contextattributes)

Looks like upstream enabled the same: https://github.com/szimek/signature_pad/pull/761#issuecomment-1986951356

**Comment by mbohovic** [comment]
> Looks like upstream enabled the same: https://github.com/szimek/signature_pad/pull/761#issuecomment-1986951356

you refer to `signature_pad` that it's fixed there, but it's useless for this package react-signature-canvas(https://github.com/agilgur5/react-signature-canvas). It is not updated or compatible with the given version of signature_pad.

Would you update the respond-signature-canvas package to be compatible with the new version of `signature_pad`?

**Comment by agilgur5** [comment]
> @agilgur5 refer to `signature_pad` that it's fixed there, but it's useless for this package [react-signature-canvas](https://github.com/agilgur5/react-signature-canvas). It is not updated

I'm very aware of that, that's why this issue is still open, and not closed.

> Would you update the react-signature-canvas package to be compatible with the new version of `signature_pad`?

There already separate issues for that, see https://github.com/agilgur5/react-signature-canvas/issues/73#issuecomment-2052280336.
